flink-core/src/main/java/org/apache/flink/core/fs/FileSystem.java

```diff
@@ -440,8 +455,18 @@ public static FileSystem getUnguardedFileSystem(final URI fsUri) throws IOExcept
 				try (TemporaryClassLoaderContext classLoaderContext = new TemporaryClassLoaderContext(classLoader)) {
 					fs = factory.create(uri);
 				}
-			}
-			else {
+			} else if (DIRECTLY_SUPPORTED_FILESYSTEM.containsKey(uri.getScheme())) {
+				final Collection<String> plugins = DIRECTLY_SUPPORTED_FILESYSTEM.get(uri.getScheme());
+				throw new UnsupportedFileSystemSchemeException(String.format(
+					"Could not find a file system implementation for scheme '%s'. The scheme is " +
+						"directly supported by Flink through the following plugin%s: %s. Please ensure that each " +
```

Comment:
```suggestion
						"directly supported by Flink through the following plugin%: %s. Please ensure that each " +
```

Reply:
This line would add an s if there multiple plugins (s3).